bokeh/properties.py

```diff
@@ -332,7 +389,14 @@ def __init__(self, delegate, help="", use_prefix=True):
 """
 
 class MetaHasProps(type):
-    def __new__(cls, class_name, bases, class_dict):
+    def __new__(meta_cls, class_name, bases, class_dict):
+        # HACK for now - we get called twice for some weird reason.
+        # __new__ used to be idempotent before PropertyDescriptor
+        # had to be transformed into Property, so it was OK, but now
+        # it isn't ... I think. Anyway, on the second call just do this?
+        # we need to figure this out for real and clean it up.
+        if '__properties__' in class_dict:
+            return type.__new__(meta_cls, class_name, bases, class_dict)
```

Comment:
Wit the following diff, tests pass and an example works on python 3, and is supposed to work on python 2.
```
diff --git a/bokeh/model.py b/bokeh/model.py
index aafd94c..ea3462d 100644
--- a/bokeh/model.py
+++ b/bokeh/model.py
@@ -3,7 +3,9 @@ from __future__ import absolute_import, print_function
 import logging
 logger = logging.getLogger(__file__)
 
-from six import add_metaclass, iteritems
+from future.utils import with_metaclass
+
+from six import iteritems
 
 from .properties import Any, HasProps, List, MetaHasProps, String
 from .query import find
@@ -64,8 +66,7 @@ class Viewable(MetaHasProps):
         else:
             raise KeyError("View model name '%s' not found" % view_model_name)
 
-@add_metaclass(Viewable)
-class Model(HasProps, CallbackManager):
+class Model(with_metaclass(Viewable, HasProps, CallbackManager)):
     """ Base class for all plot-related objects """
 
     name = String()
diff --git a/bokeh/properties.py b/bokeh/properties.py
index 611180c..b425d68 100644
--- a/bokeh/properties.py
+++ b/bokeh/properties.py
@@ -41,21 +41,23 @@ to be understood by BokehJS.
 """
 from __future__ import absolute_import, print_function
 
+import logging
+logger = logging.getLogger(__name__)
+
 import re
 import types
 import difflib
 import datetime
 import dateutil.parser
+from future.utils import with_metaclass
 import collections
 from importlib import import_module
 from copy import copy
 from warnings import warn
 import inspect
-import logging
 import numbers
-logger = logging.getLogger(__name__)
 
-from six import string_types, add_metaclass, iteritems
+from six import string_types, iteritems
 
 from . import enums
 from .util.string import nice_join
@@ -396,7 +398,8 @@ class MetaHasProps(type):
         # it isn't ... I think. Anyway, on the second call just do this?
         # we need to figure this out for real and clean it up.
         if '__properties__' in class_dict:
-            return type.__new__(meta_cls, class_name, bases, class_dict)
+            import pdb; pdb.set_trace()
+            return super(MetaHasProps, meta_cls)(meta_cls, class_name, bases, class_dict)
         names = set()
         names_with_refs = set()
         container_names = set()
@@ -555,8 +558,7 @@ def abstract(cls):
 
     return cls
 
-@add_metaclass(MetaHasProps)
-class HasProps(object):
+class HasProps(with_metaclass(MetaHasProps, object)):
 
     def __init__(self, **properties):
         super(HasProps, self).__init__()
```
I guess this is a bug in `six.add_metaclass`. Metaclass inheritance is fairly obscure.